test/functional/test_framework/mininode.py

```diff
@@ -406,17 +406,14 @@ def test_function():
 
         wait_until(test_function, timeout=timeout, lock=mininode_lock)
 
-    def wait_for_getdata(self, timeout=60):
+    def wait_for_getdata(self, hash_list, timeout=60):
         """Waits for a getdata message.
 
-        Receiving any getdata message will satisfy the predicate. the last_message["getdata"]
-        value must be explicitly cleared before calling this method, or this will return
-        immediately with success. TODO: change this method to take a hash value and only
-        return true if the correct block/tx has been requested."""
+        The object hashes in the inventory vector must match the provided hash_list."""
 
         def test_function():
             assert self.is_connected
-            return self.last_message.get("getdata")
+            return "getdata" in self.last_message and [x.hash for x in self.last_message["getdata"].inv] == hash_list
```

Comment:
nit: I'd prefer the `self.last_message.get("data")` syntax here to be consistent to other `wait_for_` methods, see e.g.:
https://github.com/bitcoin/bitcoin/blob/56d2ff8a8fc52676a22163f804a9f1a5415b4b71/test/functional/test_framework/mininode.py#L385
Or, as longer version (in terms of LOC) that only needs to access `last_message` once:
https://github.com/bitcoin/bitcoin/blob/56d2ff8a8fc52676a22163f804a9f1a5415b4b71/test/functional/test_framework/mininode.py#L392-L395

Reply:
Good idea!  Updated to use the longer version.